### PR TITLE
Add `host` option

### LINKED ISSUE #8: Node defaults to ipv6 for port resolution
We noticed that get-port is defaulting to ipv6 for port resolution, which is giving false positives. i.e. given a default port, it comes back as available, where it is actually not available. 

We will add more info to this issue shortly - Right now, we need an issue link for tracking purposes.

### PR DESCRIPTION
**Pull request by acostalima**
Resolves #8

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ getPort(3000).then(port => {
...
 ```
 
+Additionally, you can check if a preferred port is available for a given host
+
+```js
+getPort(3000, '127.0.0.1').then(port => {
+	console.log(port);
+	// Will use 3000 if available, otherwise fall back to a random port
+});
+```
```
Comment (kevva) [inline readme.md]: Just a nitpick, but there should be two blank lines betwen "major" sections.

Comment (acostalima) [inline readme.md]: @kevva Thanks! Done.

File: index.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-const getPort = port => new Promise((resolve, reject) => {
+const getPort = (port, host) => new Promise((resolve, reject) => {
 	const server = net.createServer();
 
 	server.unref();
 	server.on('error', reject);
 
-	server.listen(port, () => {
+	server.listen({port, host}, () => {
```
Comment (sindresorhus) [inline index.js]: This is not correct. `server.listen` accepts port and host as arguments, not an object.

File: index.js [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 		const port = server.address().port;
 		server.close(() => {
 			resolve(port);
 		});
 	});
 });
 
-module.exports = preferredPort => preferredPort ?
-	getPort(preferredPort).catch(() => getPort(0)) :
+module.exports = (preferredPort, preferredHost) => preferredPort ?
```
Comment (sindresorhus) [inline index.js]: It should be possible to do `getPort('127.0.0.1')` to get a random host without having a preferred port. This makes the API a bit kludgy though, so I think we should just go with an options object.

So a user can use either:

- `getPort()`
- `getPort(32)`
- `getPort({port: 32})`
- `getPort({host: '127.0.0.1'})`
- `getPort({port: 32, host: '127.0.0.1'})`

And we keep `getPort(32)` only for compatibility reasons, but don't document it.

File: readme.md [hunk 4]
```diff
@@ -30,6 +30,15 @@ getPort(3000).then(port => {
...
+
```
Comment (sindresorhus) [inline readme.md]: Can you add some API docs for the API. See https://github.com/sindresorhus/boxen#api for the style to use.

File: index.js (hunk 2 above)
Comment (acostalima) [inline index.js]: It also has an overload in which accepts an object according to the [docs](https://nodejs.org/api/net.html#net_server_listen_options_callback). I think we can go with one or another.

File: index.js (hunk 3 above)
Comment (acostalima) [inline index.js]: Indeed, it is more reasonable. I'll change that.

File: readme.md (hunk 4 above)
Comment (acostalima) [inline readme.md]: Sure. Thanks for the pointer.

File: index.js (hunk 2 above)
Comment (acostalima) [inline index.js]: I have just double checked. It is not 100% clear to me, but it seems that `host` is mandatory with the object approach which is not what we're looking for here.

Comment (sindresorhus) [inline index.js]: My fault, I was looking at the `http.listen` docs instead of `net.listen`.

**Comment by sindresorhus** [comment]
@acostalima ping :)

**Comment by sindresorhus** [comment]
Looks great. Thank you :)

**Pull request merged**